cdap-ui/app/directives/dag-plus/my-dag-ctrl.js

```diff
@@ -22,10 +22,16 @@ angular.module(PKG.name + '.commons')
     var numberFilter = $filter('number');
 
     var endpoints = [];
-    var sourceSettings = angular.copy(DAGPlusPlusFactory.getSettings(false).source);
-    var sinkSettings = angular.copy(DAGPlusPlusFactory.getSettings(false).sink);
-    var transformSourceSettings = angular.copy(DAGPlusPlusFactory.getSettings(false).transformSource);
-    var transformSinkSettings = angular.copy(DAGPlusPlusFactory.getSettings(false).transformSink);
+
+    var sourceOrigin = angular.copy(DAGPlusPlusFactory.getSettings().sourceOrigin),
```

Comment:
We should probably do something like 
```
angular.copy(DAGPlusPlusFactory.getSettings())
```
and then use objects from it. `angular.copy` is a costly operation.